docs/design/reserve-action.md

```diff
@@ -0,0 +1,152 @@
+# Reserve Action
+
+## Table of Contents
+
+- [Reserve Action](#reserve-action)
+  * [Table of Contents](#table-of-contents)
+  * [Introduction](#introduction)
+  * [Motivation](#motivation)
+  * [Function Detail](#function-detail)
+    + [Job starving](#job-starving)
+    + [Node resource reservation rule](#node-resource-reservation-rule)
+  * [Data Model](#data-model)
+  
+## Introduction
+
+The `reserve` action is to solve the problem that the task can not be scheduled for a long time. We call these long 
+unscheduled jobs starving job, if a job is not scheduled for a long time, it is a starving job. The `reserve` action 
+will try to reserve some node resources for the pods under the starving job. Nodes who has reserved resources will no 
+longer schedule new pods. When the existing pods on the node are completed, the node resources they occupy will be 
+released. Pods under the starving job will scheduled on such nodes.
+
+## Motivation
+
+When the node resources in the cluster are insufficient, the tasks with large resource requirements will be at risk of 
+starvation. The so-called starvation means that the task can not be scheduled for a long time due to various reasons.
+
+There are many reasons why a pod can not be scheduled. From the perspective of task resource requirement and cluster 
+allocatable resources, if the resources required by the task is larger that the allocatable resource of any node in the 
+cluster, the task can not be scheduled. The node under the cluster needs to be expanded vertically to increase the node 
+resources.
+
+If the node resources in the cluster has meet the resource requirements of the pod, but because of that the resources of 
+the nodes are occupied by pods with relatively small resource requirements for a long time, the task scheduling of large
+resource requirement can not be satisfied. In this case, there is no need to expand the node resources, and the `reserve` 
+action can solve such problems.
+
+The `reserve` action prevents the large resource requirement task from being starved by reserving the node resources for
+it and not allowing the other tasks to be scheduled on the node.
+
+## Function Detail 
+
+### Job starving
+
+If the `pending` time of job is greater that the set time threshold, the task has the risk of starvation, and needs to
+reserve node resources for it. If there is unscheduled task under the job, but the job has already met the `running` 
```

Comment:
Is there a risk the unscheduled tasks of the job can never be scheduled?

Reply:
In each round, the node with the most sufficient idle node is selected as the reserved node, so after several rounds of scheduling, the selected nodes will gradually converge, and finally meet the resource scheduling requirements of the task. There will be some fluctuations in this process. Fluctuation will make the task scheduled longer, and in extreme cases it will not converge, but the probability of this scene is low and can be ignored.